client/src/data.js

```diff
@@ -1,37 +1,116 @@
-let data = {
-  entries: [],
-  nextEntryId: 1,
-};
-
-window.addEventListener('beforeunload', function (event) {
-  const dataJSON = JSON.stringify(data);
-  localStorage.setItem('code-journal-data', dataJSON);
-});
-
-const localData = JSON.parse(localStorage.getItem('code-journal-data'));
-if (localData) {
-  data = localData;
-}
+import { useState, useEffect } from 'react';
 
-export function readEntries() {
-  return data.entries;
-}
+export function useCRUD() {
```

Comment:
Nice! I like how you've done this. It's similar to how I've done this in my own projects.
I like to separate out the read custom hook from the mutate custom hooks, though. It makes things a tiny bit more efficient, and that's what React Query and useSWR do, so it's easier to work with them that way.